itest/psbt_test.go

```diff
@@ -1265,6 +1267,162 @@ func testMultiInputPsbtSingleAssetID(t *harnessTest) {
+		bob      = secondTapd
+		numUnits = uint64(500)
+	)
+
+	// We need to derive two keys, one for the new script key and one for
+	// the internal key.
+	bobScriptKey, bobInternalKey := deriveKeys(t.t, bob)
+
+	// Now we create a script tree consisting of two simple scripts.
+	preImage := []byte("hash locks are cool")
+	leaf1 := test.ScriptHashLock(t.t, preImage)
+	leaf2 := test.ScriptSchnorrSig(t.t, bobScriptKey.RawKey.PubKey)
+	leaf1Hash := leaf1.TapHash()
+	leaf2Hash := leaf2.TapHash()
+	tapscript := input.TapscriptPartialReveal(
+		bobScriptKey.RawKey.PubKey, leaf2, leaf1Hash[:],
+	)
+	rootHash := tapscript.ControlBlock.RootHash(leaf2.Script)
+
+	sendToTapscriptAddr(
+		ctxt, t, alice, bob, numUnits, genInfo, mintedAsset,
+		bobScriptKey, bobInternalKey, tapscript, rootHash,
+	)
+
+	// Now try to send back those assets using the PSBT flow.
+	aliceAddr, err := alice.NewAddr(ctxb, &taprpc.NewAddrRequest{
+		AssetId:      genInfo.AssetId,
+		Amt:          numUnits / 5,
+		AssetVersion: mintedAsset.Version,
+	})
+	require.NoError(t.t, err)
+	AssertAddrCreated(t.t, alice, rpcAssets[0], aliceAddr)
+
+	fundResp := fundAddressSendPacket(t, bob, aliceAddr)
+	t.Logf("Funded PSBT: %v",
+		base64.StdEncoding.EncodeToString(fundResp.FundedPsbt))
+
+	fundedPacket, err := tappsbt.NewFromRawBytes(
+		bytes.NewReader(fundResp.FundedPsbt), false,
+	)
+	require.NoError(t.t, err)
+
+	// We can now ask the wallet to sign the script path, since we only need
+	// a signature.
+	controlBlockBytes, err := tapscript.ControlBlock.ToBytes()
+	require.NoError(t.t, err)
+	fundedPacket.Inputs[0].TaprootMerkleRoot = rootHash[:]
+	fundedPacket.Inputs[0].TaprootLeafScript = []*psbt.TaprootTapLeafScript{
+		{
+			ControlBlock: controlBlockBytes,
+			Script:       leaf2.Script,
+			LeafVersion:  leaf2.LeafVersion,
+		},
+	}
+	fundedPacket.Inputs[0].TaprootBip32Derivation[0].LeafHashes = [][]byte{
+		leaf2Hash[:],
+	}
+
+	// Before signing, we set the sighash of the first input to SIGHASH_NONE
+	// which allows us to alter the outputs of the PSBT after the signature
+	// has been generated.
+	fundedPacket.Inputs[0].SighashType = txscript.SigHashNone
+
+	var b bytes.Buffer
+	err = fundedPacket.Serialize(&b)
+	require.NoError(t.t, err)
+
+	fmt.Println("SGHSH: bob signing")
+	signedResp, err := bob.SignVirtualPsbt(
+		ctxb, &wrpc.SignVirtualPsbtRequest{
+			FundedPsbt: b.Bytes(),
+		},
+	)
+	require.NoError(t.t, err)
+	require.Contains(t.t, signedResp.SignedInputs, uint32(0))
+	fmt.Println("SGHSH: bob signing done")
+
+	// Now we deserialize the signed packet again in order to edit it
+	// and then anchor it.
+	signedPacket, err := tappsbt.NewFromRawBytes(
+		bytes.NewReader(signedResp.SignedPsbt), false,
+	)
+	require.NoError(t.t, err)
+
+	const (
+		magicAmtOverride = 42
+	)
+
+	// Edit the already signed PSBT and change the VOutput amount. This
+	// should be ok as we used SIGHASH_NONE for the generated signature.
+	signedPacket.Outputs[1].Asset.Amount = magicAmtOverride
```

Comment:
this is probably wrong, overriding the outputs should also update the root split commitment as that's where the amount inflation/deflation is checked

Reply:
IIUC you removed that check in the diff. 

What we want here is:
  * Alice signs input off to Bob, she doesn't know Bob's script key yet. 
  * Bob takes the signature, places it in a Bitcoin output commitment he controls, then swaps in his actual script key. 
  * Bob can now broadcast a state transition that transfers that asset to him.

The other portion is on the Bitcoin sighash side:
   * Returning to the above, Alice's prev asset outpoint must be spent for that transaction to be valid (Alice's prev input points to that prev outpoint). 
   * Alice signs the PSBT that has 1 BTC input, and 2 BTC output. This needs Bob to provide that 1 BTC input. This uses sighash single + anyonecanpay on the Bitcoin side. 
   * Bob can then take the sig, make a new commitment, attach his input, then broadcast the Bitcoin tx (signs with sighash all).

Reply:
> IIUC you removed that check in the diff.

That was part of the closed PR. Here I am not removing any supply checks

> What we want here is:

Guess you're describing the swap flow? This itest is only about a simple check that sighashes work. The trustless swap will be a separate test.